staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/plugins.go

```diff
@@ -27,8 +27,13 @@ import (
 	"github.com/golang/glog"
 
 	"k8s.io/apiserver/pkg/storage/value/encrypt/envelope"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/envelope/vault"
 )
 
+func init() {
+	KMSPluginRegistry.Register("vault", vault.KMSFactory)
```

Comment:
@jhorwit2 Thanks for reviewing the code, it was very helpful, I have made most of the changes suggested by you. Does the PR look good to you? Once internal review is over, I will create a new branch which has only 2 commits (one for code changes, one for dependencies and build) sync with latest upstream master and run all local verify/test/integration-test again.